filters/tee/tee_test.go

```diff
@@ -136,17 +152,76 @@ func TestTeeEndToEndBody(t *testing.T) {
 	defer p.Close()
 
 	testingStr := "TESTEST"
-	req, _ := http.NewRequest("GET", p.URL, strings.NewReader(testingStr))
+	req, err := http.NewRequest("GET", p.URL, strings.NewReader(testingStr))
+	if err != nil {
+		t.Error(err)
+	}
+
 	req.Host = "www.example.org"
-	req.Header.Set("X-Test", "true")
```

Comment:
it didn't hurt there but there was no assertion for it. I wrote a new test case for checking the headers.